src/qt/bitcoingui.cpp

```diff
@@ -889,16 +889,26 @@ void BitcoinGUI::createIconMenu(QMenu *pmenu)
     // Configuration of the tray icon (or dock icon) icon menu
     pmenu->addAction(toggleHideAction);
     pmenu->addSeparator();
-    pmenu->addAction(sendCoinsMenuAction);
-    pmenu->addAction(coinJoinCoinsMenuAction);
-    pmenu->addAction(receiveCoinsMenuAction);
-    pmenu->addSeparator();
-    pmenu->addAction(signMessageAction);
-    pmenu->addAction(verifyMessageAction);
+
+    if (enableWallet)
+    {
+        pmenu->addAction(sendCoinsMenuAction);
+        pmenu->addAction(coinJoinCoinsMenuAction);
+        pmenu->addAction(receiveCoinsMenuAction);
+        pmenu->addSeparator();
+        pmenu->addAction(signMessageAction);
+        pmenu->addAction(verifyMessageAction);
+    }
+
     pmenu->addSeparator();
     pmenu->addAction(optionsAction);
     pmenu->addAction(openInfoAction);
-    pmenu->addAction(openRPCConsoleAction);
+
+    if (enableWallet)
+    {
+        pmenu->addAction(openRPCConsoleAction);
```

Comment:
why not just move this into the above if block?

Reply:
I had to split because of the separator, that would change the order of things when the wallet was enabled.

Reply:
Bitcoin had the seperator too https://github.com/bitcoin/bitcoin/pull/14383/files#diff-2ecf8cbf369cf3d2f3d2b1cf5cfe4c1a647d63e11e2885d2fd0ac11fb5f7a804R629

I think you can move this (and the seperator) into the above if block as bitcoin did.